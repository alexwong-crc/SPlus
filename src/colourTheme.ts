export default {
  primary: {
    main: { code: "#214464", text: "#FFFFFF" },
    light: { code: "#3777B2", text: "#FFFFFF" },
    dark: { code: "#0C2134", text: "#FFFFFF" }
  },
  secondary: {
    main: { code: "#C14050", text: "#FFFFFF" },
    light: { code: "#FA485E", text: "#FFFFFF" },
    dark: { code: "#74212B", text: "#FFFFFF" }
  },
  complimentary: {
    main: { code: "#F1F45A", text: "#2C2C2C" },
    light: { code: "#FAFC97", text: "#2C2C2C" },
    dark: { code: "#DBDF16", text: "#2C2C2C" }
  },
  action: {
    main: { code: "#76C56F", text: "#FFFFFF" },
    light: { code: "#A0F199", text: "#FFFFFF" },
    dark: { code: "#32AA27", text: "#FFFFFF" }
  },
  error: {
    main: { code: "#FF001F", text: "#FFFFFF" },
    light: { code: "#FF4E4E", text: "#FFFFFF" },
    dark: { code: "#CB0018", text: "#FFFFFF" }
  },
  black: {
    main: { code: "#2C2C2C", text: "#FFFFFF" },
    shadeOne: { code: "#454545", text: "#FFFFFF" },
    shadeTwo: { code: "#696767", text: "#FFFFFF" },
    shadeThree: { code: "#979797", text: "#FFFFFF" },
    shadeFour: { code: "#CCCCCC", text: "#2C2C2C" },
    shadeFive: { code: "#ECECEC", text: "#2C2C2C" },
    shadeSix: { code: "#F8F8F8", text: "#2C2C2C" }
  }
};

interface IColourCode {
  code: string;
  text: string;
}

interface IColour {
  main: IColourCode;
  light: IColourCode;
  dark: IColourCode;
}

interface IColourBlack {
  main: IColourCode;
  shadeOne: IColourCode;
  shadeTwo: IColourCode;
  shadeThree: IColourCode;
  shadeFour: IColourCode;
  shadeFive: IColourCode;
  shadeSix: IColourCode;
}

export interface IColourTheme {
  primary: IColour;
  secondary: IColour;
  complimentary: IColour;
  action: IColour;
  error: IColour;
  black: IColourBlack;
}
