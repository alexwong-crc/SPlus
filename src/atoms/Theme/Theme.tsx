import React from "react";
import { ThemeProvider } from "styled-components";
import colourTheme from "../../colourTheme";

interface ITheme {
  [index: string]: string | object;
}

interface IProps {
  children: React.ReactNode;
  theme: ITheme;
}

const Theme: React.SFC<IProps> = ({
  children,
  theme = colourTheme
}: IProps) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

export default Theme;
