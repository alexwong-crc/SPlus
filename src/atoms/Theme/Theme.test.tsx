import React from "react";
import { shallow } from "enzyme";
import Theme from "./Theme";
import colourTheme from "../../styles/colourTheme";

describe("Theme component", () => {
  it("should match snapshot", () => {
    const component = shallow(
      <Theme>
        <div>hello</div>
      </Theme>
    );

    expect(component).toMatchSnapshot();
  });

  it("should render children components", () => {
    const component = shallow(
      <Theme>
        <div>hello</div>
      </Theme>
    );

    expect(component.contains(<div>hello</div>)).toBeTruthy();
  });

  it("should have a default colour theme when it is not passed", () => {
    const component = shallow(<Theme>Hello</Theme>);

    expect(component.props().theme).toMatchObject(colourTheme);
  });

  it("should take in a new theme when passed", () => {
    const testTheme = {
      primary: "bob"
    };
    const component = shallow(<Theme theme={testTheme}>Hello</Theme>);

    expect(component.props().theme).toMatchObject(testTheme);
  });
});
