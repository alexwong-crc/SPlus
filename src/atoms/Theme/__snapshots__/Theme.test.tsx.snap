// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme component should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "action": Object {
        "dark": Object {
          "code": "#32AA27",
          "text": "#FFFFFF",
        },
        "light": Object {
          "code": "#A0F199",
          "text": "#FFFFFF",
        },
        "main": Object {
          "code": "#76C56F",
          "text": "#FFFFFF",
        },
      },
      "black": Object {
        "main": Object {
          "code": "#2C2C2C",
          "text": "#FFFFFF",
        },
        "shadeFive": Object {
          "code": "#ECECEC",
          "text": "#2C2C2C",
        },
        "shadeFour": Object {
          "code": "#CCCCCC",
          "text": "#2C2C2C",
        },
        "shadeOne": Object {
          "code": "#454545",
          "text": "#FFFFFF",
        },
        "shadeSix": Object {
          "code": "#F8F8F8",
          "text": "#2C2C2C",
        },
        "shadeThree": Object {
          "code": "#979797",
          "text": "#FFFFFF",
        },
        "shadeTwo": Object {
          "code": "#696767",
          "text": "#FFFFFF",
        },
      },
      "complimentary": Object {
        "dark": Object {
          "code": "#DBDF16",
          "text": "#2C2C2C",
        },
        "light": Object {
          "code": "#FAFC97",
          "text": "#2C2C2C",
        },
        "main": Object {
          "code": "#F1F45A",
          "text": "#2C2C2C",
        },
      },
      "error": Object {
        "dark": Object {
          "code": "#CB0018",
          "text": "#FFFFFF",
        },
        "light": Object {
          "code": "#FF4E4E",
          "text": "#FFFFFF",
        },
        "main": Object {
          "code": "#FF001F",
          "text": "#FFFFFF",
        },
      },
      "primary": Object {
        "dark": Object {
          "code": "#0C2134",
          "text": "#FFFFFF",
        },
        "light": Object {
          "code": "#3777B2",
          "text": "#FFFFFF",
        },
        "main": Object {
          "code": "#214464",
          "text": "#FFFFFF",
        },
      },
      "secondary": Object {
        "dark": Object {
          "code": "#74212B",
          "text": "#FFFFFF",
        },
        "light": Object {
          "code": "#FA485E",
          "text": "#FFFFFF",
        },
        "main": Object {
          "code": "#C14050",
          "text": "#FFFFFF",
        },
      },
    }
  }
>
  <div>
    hello
  </div>
</ThemeProvider>
`;
